joint_trajectory_controller/src/joint_trajectory_controller.cpp

```diff
@@ -191,39 +191,48 @@ controller_interface::return_type JointTrajectoryController::update(
       const bool before_last_point = end_segment_itr != (*traj_point_active_ptr_)->end();
 
       // set values for next hardware write()
+      if (use_closed_loop_pid_adapter)
+      {
+        // Update PIDs
+        for (auto i = 0ul; i < joint_num; ++i)
+        {
+          tmp_command_[i] = (state_desired.velocities[i] * ff_velocity_scale_[i]) +
+                            pids_[i]->computeCommand(
+                              state_desired.positions[i] - state_current.positions[i],
+                              state_desired.velocities[i] - state_current.velocities[i],
+                              (uint64_t)period.nanoseconds());
+        }
+      }
       if (has_position_command_interface_)
       {
         assign_interface_from_point(joint_command_interface_[0], state_desired.positions);
       }
       if (has_velocity_command_interface_)
       {
-        if (!use_closed_loop_pid_adapter)
+        if (use_closed_loop_pid_adapter)
```

Comment:
This set of cascading `if`-s should be refactored into something more readable.
I suggest 2 things:
* make the selection of choosing the command somehow more explicit with different variable names, etc: `tmp_command_` is not very intuitive
* refactor all the `assign_interface_from_point` uses into a function that gets a parameter, the command (whether it's desired state or the pid command)